src/Graphics/Vty/Attributes/Color.hs

```diff
@@ -125,12 +126,15 @@ detectColorMode termName' = do
                   (\(_ :: Terminfo.SetupTermError) -> return Nothing)
     let getCap cap = term >>= \t -> Terminfo.getCapability t cap
         termColors = fromMaybe 0 $ getCap (Terminfo.tiGetNum "colors")
+    colorterm <- lookupEnv "COLORTERM"
     return $ if
-        | termColors <  8   -> NoColor
-        | termColors <  16  -> ColorMode8
-        | termColors == 16  -> ColorMode16
-        | termColors <  256 -> ColorMode240 (fromIntegral termColors - 16)
-        | otherwise         -> ColorMode240 240
+        | termColors <  8               -> NoColor
+        | termColors <  16              -> ColorMode8
+        | termColors == 16              -> ColorMode16
+        | termColors <  256             -> ColorMode240 (fromIntegral termColors - 16)
+        | colorterm == Just "truecolor" -> FullColor
```

Comment:
These checks assume that the value of `termColors` obtained from `terminfo` will always be larger than 256 if the environment variable is set. Is that a safe assumption?

Reply:
I think it is a safe assumption. The value `termColors` is based on `TERM` environment variable. See the testing I did for more details. For most terminals it seems `termColors == 256` and they support at least `ColorMode240`. It seems some terminals do not advertise their full capabilities with the environment variables `TERM` and `COLORTERM`. I am happy to switch the order of the checks. I think the code as it is makes more sense though, especially in the case when `COLORTERM` is set for whatever reason and not overridden by the terminal emulator.

Reply:
Okay, thanks for the extra details.